src/App.tsx

```diff
@@ -1,15 +1,64 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { setTimeout } from 'timers';
 import './App.scss';
+
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [isInputFocus, setIsInputFocus] = useState(false);
+
+  const { name, born, died } = selectPerson || {};
+
+  const selectedPersonInfo = selectPerson
+    ? `${name} (${born} - ${died})`
+    : 'No selected person';
+
+  const handlePesonClick = (person: Person, namePerson: string) => {
+    setSelectPerson(person);
+    setQuery(namePerson);
+    setIsInputFocus(false);
+  };
+
+  const applyQuery = useMemo(
+    () => debounce((value: string) => {
+      setAppliedQuery(value);
+      setIsInputFocus(true);
+    }, 900),
+    [],
+  );
+
+  const queryInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setSelectPerson(null);
+    setIsInputFocus(false);
+  };
+
+  const handleInputFocus = () => {
+    setIsInputFocus(true);
+  };
+
+  const handleInputBlur = () => {
+    setTimeout(() => setIsInputFocus(false), 100);
+  };
+
+  const currentPerson = useMemo(() => {
```

Comment:
```suggestion
  const filteredPeople = useMemo(() => {
```
it's better name - just FYI